src/main/java/seedu/address/model/person/DetailsContainsPredicate.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.model.person;
+
+import java.util.Iterator;
+import java.util.function.Predicate;
+
+import seedu.address.logic.commands.FindCommand.FindDetailDescriptor;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Tests that every {@code ReadOnlyPerson}'s attribute matches corresponding keyword
+ * in descriptor given except it is null.
+ */
+public class DetailsContainsPredicate implements Predicate<ReadOnlyPerson> {
+    private FindDetailDescriptor descriptor;
+
+    public DetailsContainsPredicate(FindDetailDescriptor descriptor) {
+        this.descriptor = descriptor;
+    }
+
+    @Override
+    public boolean test(ReadOnlyPerson person) {
+        if (descriptor.getName().isPresent()
+                && !person.getName().fullName.toLowerCase().contains(
+                descriptor.getName().get().toLowerCase())) {
```

Comment:
Double check if these code violates our coding standard or not (Section. `Place line break to improve readability`)